eventdata/runners/kibana_runner.py

```diff
@@ -33,12 +32,12 @@ def kibana(es, params):
     response['unit'] = "ops"
     response['visualisation_count'] = visualisations
 
-    if logger.isEnabledFor(logging.DEBUG):
-        logger.debug("[kibana_runner] request: {}".format(request))
-
-    result = es.msearch(body = request)
+    if "pre_filter_shard_size" in meta_data:
+        result = es.msearch(body = request, params = {'pre_filter_shard_size': meta_data['pre_filter_shard_size']})
+    else:
+        result = es.msearch(body = request)
 
-    if logger.isEnabledFor(logging.DEBUG):
-        logger.debug("[kibana_runner] result: {}".format(result))
+    if 'debug' in params['meta_data'] and params['meta_data']['debug']:
```

Comment:
I think `debug` should also be a mandatory parameter. This simplifies the conditional a little bit as well.